src/transformers/models/distilbert/modeling_distilbert.py

```diff
@@ -255,7 +259,11 @@ class TransformerBlock(nn.Module):
     def __init__(self, config: PretrainedConfig):
         super().__init__()
 
-        assert config.dim % config.n_heads == 0
+        if self.dim % self.n_heads != 0:
```

Comment:
```suggestion
        if config.dim % config.n_heads != 0:
```
I think this is the cause of the issue.